src/swap.spec.ts

```diff
@@ -0,0 +1,36 @@
+// import { Problem } from "../dist/src";
+import nock = require("nock");
+import { Cnd } from "./cnd/cnd";
+import { Swap } from "./swap";
+
+describe("Swap", () => {
+  it("Throws a problem if cnd returns an error when executing an action", async () => {
+    const basePath = "http://example.com";
+    const cnd = new Cnd(basePath);
+
+    const selfPath = "/swap-id";
+
+    const swap = new Swap(cnd, selfPath, {});
+
+    const scope = nock(basePath)
+      .get(selfPath)
+      .reply(400, JSON.stringify({}), {
+        "content-type": "application/vnd.siren+json"
+      });
+
+    const promise = swap.tryExecuteSirenAction("action", {
+      maxTimeoutSecs: 1,
+      tryIntervalSecs: 0.01
+    });
+
+    try {
+      await promise;
+    } catch (e) {
+      console.log(JSON.stringify(e, undefined, 2));
```

Comment:
@thomaseizinger this `e` is not a `Problem` but an `Error`. I have done another tests that check the instance of it and it fails too. The following is printed here:

```
 {
      "message": "Request failed with status code 400",
      "name": "Error",
      "stack": "Error: Request failed with status code 400\n    at createError (/Users/froyer/src/comit-network/comit-js-sdk/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/Users/froyer/src/comit-network/comit-js-sdk/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/Users/froyer/src/comit-network/comit-js-sdk/node_modules/axios/lib/adapters/http.js:236:11)\n    at IncomingMessage.emit (events.js:311:20)\n    at endReadableNT (_stream_readable.js:1204:12)\n    at processTicksAndRejections (internal/process/task_queues.js:84:21)",
      "config": {
        "url": "/swap-id",
        "method": "get",
        "headers": {
          "Accept": "application/json, text/plain, */*",
          "User-Agent": "axios/0.19.2"
        },
        "baseURL": "http://example.com",
        "transformRequest": [
          null
        ],
        "transformResponse": [
          null
        ],
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1
      }
    }
```

I was expecting a problem to be thrown by `Cnd.fetch`. What am I missing?

Reply:
Ignore me!